src/directConnectManager.test.ts

```diff
@@ -526,4 +540,46 @@ describe("mergeSecrets", () => {
     const finalSchemaSecret = (result.schema_registry?.credentials as any)?.api_secret;
     assert.equal(finalSchemaSecret, "newApiSecret");
   });
+
+  it("should replace placeholder SCRAM auth passwords with current password", () => {
+    const newScramAuthSpec: CustomConnectionSpec = {
```

Comment:
Minor: we probably want to use [`CustomConnectionSpecFromJSON`](https://github.com/confluentinc/vscode/blob/68678cedcb6415f84dbf551b1812b65a26008982/src/storage/resourceManager.ts#L643) in these to make sure we aren't missing any fields/types with all the nested `FromJSON` calls